### ISSUE TITLE
infer uberState type

### ISSUE THREAD

**Issue by SiriusAshling** [sparkle-preference/OriWotwRandomizerClient#187]
Investigate how simple it would be to infer the bool/teleporter/byte/int/float type when setting uberStates (e.g. `8|9|0|int|1` -> `8|9|0|1`)

This would remove the need to check things such as whether an uberState is a byte or an int.
It would also make header language even more number oriented which is definitely desirable.

**Comment by MunWolf** [comment]
Does the seedgen need these types?
As we can easily get the type of uberstate client side.

**Comment by SiriusAshling** [comment]
I don't think it strictly needs them, but it might be good to have them in order to throw certain errors already during generation time instead of during runtime.
